Documentation/proposals/202501-configuration-object-status-subresource.md

```diff
@@ -0,0 +1,193 @@
+            reason: ReconcileSucceeded
+            message: "Successfully reconciled with Prometheus"
+        pods:
+          - name: prometheus-main-0
+            namespace: monitoring
+            endpoint: http://10.42.0.10:9090
+            totalScrapedTargets: 3
+            totalUpTargets: 2
+            totalDownTargets: 1
+            lastCheckedTime: "2025-05-20T12:34:56Z"
+```
+
+CRDs of podMonitor, scrapeCOnfig and Probes looks similar to above.
+
+The operator sends the GET request at regular interval to the config-reloader sidecar with the serviceMonitor selector labels as the request body, the sidecar container after receiving the request sends the /api/v1/targets request to prometheus-container, from the response it gets from the prometheus, it modifies the response based on the labels and send the response to the operator. The sidecar will remove information from the response which can be used by attackers.
+
+CRD of PrometheusRule
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: example-prometheus-rules
+  namespace: monitoring
+  labels:
+    prometheus: k8s
+    role: alert-rules
+spec:
+  groups:
+    - name: example.rules
+      interval: 30s
+      rules:
+        - alert: HighPodCPUUsage
+          expr: sum(rate(container_cpu_usage_seconds_total{container!="", pod!=""}[5m])) by (pod) > 0.5
+          for: 5m
+          labels:
+            severity: warning
+          annotations:
+            summary: "High CPU usage on pod {{ $labels.pod }}"
+            description: "Pod {{ $labels.pod }} is using more than 0.5 cores for 5 minutes."
+  status:
+    bindings:
+      - resource: prometheus
+        name: prometheus-main
+        namespace: monitoring
+        conditions:
+          - type: Reconciled
+            status: "True"
+            observedGeneration: 1
+            lastTransitionTime: "2025-05-20T12:34:56Z"
+            reason: ReconcileSucceeded
+            message: "Successfully reconciled with Prometheus"
+```
+
+CRD of AlertManagerConfig
+
+```yaml
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: AlertmanagerConfig
+metadata:
+  name: minimal-alertmanager-config
+  namespace: monitoring
+spec:
+  route:
+    receiver: "webhook-receiver"
+  receivers:
+    - name: "webhook-receiver"
+      webhookConfigs:
+        - url: "http://my-webhook-service.monitoring.svc:8080/"
+          sendResolved: true
+  status:
+    bindings:
+      - resource: alertmanager
+        name: alertmanager-main
+        namespace: monitoring
+        conditions:
+          - type: Reconciled
+            status: "True"
+            observedGeneration: 1
+            lastTransitionTime: "2025-05-20T12:34:56Z"
+            reason: ReconcileSucceeded
+            message: "Successfully reconciled with Prometheus"
+```
+
+Finalizers are used during the deletion of the config-resource and workload-resource to clear the refrences in the status-subresource.
```

Comment:
(nit) That is correct. But I don't see why it needs to be mentioned here

Reply:
IMHO details about why we need finalizers and how it will work could be its own section.